include/swift/AST/DiagnosticsSema.def

```diff
@@ -2730,6 +2730,10 @@ ERROR(escaping_non_function_parameter,none,
 NOTE(escaping_optional_type_argument, none,
      "closure is already escaping in optional type argument", ())
 
+// @_nonEphemeral attribute
+ERROR(non_ephemeral_non_pointer_type,none,
+      "@_nonEphemeral attribute currently only applies to pointer types", ())
```

Comment:
I don't know if it's a good idea to put "currently" in a diagnostic if we have no concrete plans to do anything else.